### PR TITLE
Add bindings for FreeBSD's ifaddrs.h.

### PR DESCRIPTION
**Pull request by jmdavis**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: druntime/src/core/sys/freebsd/ifaddrs.d [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/++
+    D header file for FreeBSD's ifaddrs.h.
+
+    Copyright: Copyright 2023
+    License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+    Authors:   $(HTTP jmdavisprog.com, Jonathan M Davis)
+ +/
+module core.sys.freebsd.ifaddrs;
+
+import core.sys.posix.sys.socket : sockaddr;
```
Comment (thewilsonator) [inline druntime/src/core/sys/freebsd/ifaddrs.d:12]: this should be below the `version (FreeBSD):`, the selective import causes failures on non-posix systems (i.e. Windows)

Comment (jmdavis) [reply druntime/src/core/sys/freebsd/ifaddrs.d]: fixed

**Comment by thewilsonator** [comment]
Did you want to add a changelog entry for this?

**Comment by jmdavis** [comment]
I don't really care. It's been a while since I committed any bindings, and I don't recall what we normally do. I can add one if we want to have one.

**Pull request merged**